frontend/packages/dev-console/src/components/project-access/__tests__/projectAccessForm-utils.spec.tsx

```diff
@@ -0,0 +1,406 @@
+import { cloneDeep } from 'lodash';
+import {
+  validationSchema,
+  getRolesWithNameChange,
+  UserRoleBinding,
+  getRemovedRoles,
+  getNewRoles,
+  getFinalRoles,
+  filterRoleBindings,
+  Roles,
+  getUserRoleBindings,
+} from '../projectAccessForm-utils';
+
+const mockProjectAccessData = {
+  projectAccess: [
+    {
+      user: 'abc',
+      role: 'view',
+    },
+  ],
+};
+describe('ValidationUtils', () => {
+  it('should throw an error if Name field is empty', async () => {
+    const mockData = cloneDeep(mockProjectAccessData);
+    mockData.projectAccess[0].user = '';
+
+    await validationSchema.isValid(mockData).then((valid) => expect(valid).toEqual(false));
+    await validationSchema.validate(mockData).catch((err) => {
+      expect(err.message).toBe('Required');
+      expect(err.type).toBe('required');
+    });
+  });
+
+  it('should throw an error if no Role is selected ', async () => {
+    const mockData = cloneDeep(mockProjectAccessData);
+    mockData.projectAccess[0].role = '';
+
+    await validationSchema.isValid(mockData).then((valid) => expect(valid).toEqual(false));
+    await validationSchema.validate(mockData).catch((err) => {
+      expect(err.message).toBe('Required');
+      expect(err.type).toBe('required');
+    });
+  });
+});
+
+describe('Fetch required roles', () => {
+  it('should fetch the only the required rolebindings', async () => {
+    const roleBindings = {
```

Comment:
Move all of the mock data out into a separate mock file. Change that data in each test as required.